nornir/plugins/inventory/netbox.py

```diff
@@ -270,14 +230,40 @@ def __init__(
                     host.get("data").get("config_context").get("nornir").get("port")
                 )
 
-            # Attempt to add 'platform' based of value in 'slug'
-            host["platform"] = dev["platform"]["slug"] if dev["platform"] else None
-
             # Assign temporary dict to outer dict
             # Netbox allows devices to be unnamed, but the Nornir model does not allow this
             # If a device is unnamed we will set the name to the id of the device in netbox
             hosts[dev.get("name") or dev.get("id")] = host
 
+    @classmethod
+    def _get_paginated(cls, nb_url, session):
+        url = f"{nb_url}/api/secrets/secrets/?limit=0"
```

Comment:
Personally I would just take an argument "url" in the function definition, which should be a constructed url to an api endpoint.
L240 can then be deleted.

Reply:
I think I have cleared all of these up.

Reply:
tests are updated and working.